src/components/AppNavigationItem/AppNavigationItem.vue

```diff
@@ -19,93 +20,253 @@
+
+```
+<AppNavigationItem title="Item with counter" icon="icon-folder">
+	<AppNavigationCounter slot="counter">
+		99+
+	</AppNavigationCounter>
+</AppNavigationItem>
+```
+
+### Element with children
+
+Wrap the children in a template with the `slot` property and use the prop `allowCollapse` to choose wether to allow or
+prevent the user from collapsing the items.
+
+```
+<AppNavigationItem title="Item with children" :allowCollapse="true">
+	<template>
+		<AppNavigationItem title="AppNavigationItemChild1" />
+		<AppNavigationItem title="AppNavigationItemChild2" />
+		<AppNavigationItem title="AppNavigationItemChild3"  />
+		<AppNavigationItem title="AppNavigationItemChild4"  />
+	</template>
+</AppNavigationItem>
+```
+### Editable element
+Add the prop `:editable=true` and an edit placeholder if you need it. By devault
+the placeholder is the previous title of the element.
+
+```
+<AppNavigationItem title="Editable Item" :editable="true"
+	editPlaceholder="your_placeholder_here" />
+```
+### Undo element
+Just set the `undo` and `title` props.
+
+```
+<AppNavigationItem :undo="true" title="Are you sure you want to delete?"  />
 
+```
+### Pinned element
+Just set the `pinned` prop.
+```
+<AppNavigationItem title="Pinned item" :pinned="true" />
+```
+
+</docs>
+
+<template>
+	<!-- Navigation item, can be either an <li> or a <router-link> depending on the props -->
+	<nav-element v-bind="navElement"
+		:title="title"
+		:class="{
+			'app-navigation-entry--no-icon': !isIconShown,
+			'app-navigation-entry--opened': opened,
+			'app-navigation-entry--pinned': pinned,
+			'app-navigation-entry--editing' : editing,
+			'app-navigation-entry--deleted': undo,
+			'app-navigation-entry--collapsible': collapsible
+		}"
+		class="app-navigation-entry">
+		<!-- Icon and title -->
+		<a v-if="!undo && !editing"
+			class="app-navigation-entry-link" href="#"
+			@click="onClick">
+
+			<!-- icon if not collapsible -->
+			<!-- never show the icon over the collapsible if mobile -->
+			<div :class="{ 'icon-loading-small': loading, [icon]: icon && isIconShown }"
+				class="app-navigation-entry-icon">
+				<slot v-if="!loading" v-show="isIconShown" name="icon" />
+			</div>
+			{{ title }}
+		</a>
+
+		<AppNavigationIconCollapsible v-if="collapsible" :open="opened" @click.prevent.stop="toggleCollapse" />
 		<!-- undo entry -->
-		<div v-if="item.undo" class="app-navigation-entry-deleted">
-			<div class="app-navigation-entry-deleted-description">
-				{{ item.undo.text }}
+		<div v-if="undo" class="app-navigation-entry__deleted">
+			<div class="app-navigation-entry__deleted-description">
+				{{ title }}
 			</div>
-			<button :title="t('settings', 'Undo')" class="app-navigation-entry-deleted-button icon-history" />
+		</div>
+
+		<!-- Counter and Actions -->
+		<div v-if="hasUtils" class="app-navigation-entry__utils">
+			<slot name="counter" />
+			<Actions menu-align="right" :open.sync="openedMenu">
```

Comment:
Sorry, my fault: `openedMenu` isn't passed automatically to the parent component. We have to use `menuOpen` since there is a watcher on it.
```suggestion
			<Actions menu-align="right" :open.sync="menuOpen">
```

Reply:
yes, watcher and stuff are not used anymore.
Actions already manage its own internal state.

Though what you suggests is not valid in vue. You cannot edit a prop directly. You need to trigger an event to update it :)